### PR TITLE
Add support for attachments api

### PR DESCRIPTION
**Pull request by yifanplanet**
- Add support for attachments api

[AV-2766](https://nylas.atlassian.net/browse/AV-2766)
[AV-2767](https://nylas.atlassian.net/browse/AV-2767)

# License
I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

[AV-2766]: https://nylas.atlassian.net/browse/AV-2766?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AV-2767]: https://nylas.atlassian.net/browse/AV-2767?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/models/attachments.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  /**
+   * ID of the message the attachment belongs to.
+   */
+  messageId: string;
+}
+
+/**
+ * Interface representing of the query parameters for downloading an attachment.
+ */
+export type DownloadAttachmentQueryParams = Partial<FindAttachmentQueryParams>;
```
Comment (mrashed-dev) [inline src/models/attachments.ts:52]: This shouldn't be a partial, messageId is required in both attachment calls. 

```suggestion
export type DownloadAttachmentQueryParams = FindAttachmentQueryParams;
```

File: src/resources/attachments.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+ * You can use the attachments schema in a Send v3 request to send attachments regardless of the email provider.
+ * To include attachments in a Send v3 request, all attachments must be base64 encoded and the encoded data must be placed in the content field in the request body.
+ *
+ * If you are using draft support, the draft including the attachment is stored on the provider. If you are not using draft support, Nylas stores the attachment.
+ *
+ * Attachment size is currently limited by each provider.
+ * 3MB for Gmail messages
+ * 10MB for Microsoft messages
+ *
+ * You can also send attachments inline in an email message, for example for images that should be displayed in the email body content.
```
Comment (mrashed-dev) [inline src/resources/attachments.ts:46]: nit: I think this is not needed

**Review by mrashed-dev — CHANGES_REQUESTED** [review]
Awesome work @yifanplanet! Left some comments.

**Comment by yifanplanet** [comment]
> Awesome work @yifanplanet! Left some comments.

@mrashed-dev Thanks for the review! Updates were made based on the requests.

**Pull request merged**